{
  "id": "GHSA-2mw7-wggm-m6w3",
  "summary": "Denial of Service in ethereumjs-vm",
  "details": "ethereumjs-vm 2.4.0 allows attackers to cause a denial of service (vm.runCode failure and REVERT) via a \"code: Buffer.from(my_code, 'hex')\" attribute.",
  "aliases": [
    "CVE-2018-19183"
  ],
  "modified": "2022-09-14T22:05:59Z",
  "published": "2018-11-21T22:22:46Z",
  "database_specific": {
    "nvd_published_at": null,
    "github_reviewed_at": "2020-06-16T20:52:41Z",
    "severity": "HIGH",
    "github_reviewed": true,
    "cwe_ids": [
      "CWE-119"
    ]
  },
  "references": [
    {
      "type": "ADVISORY",
      "url": "https://nvd.nist.gov/vuln/detail/CVE-2018-19183"
    },
    {
      "type": "WEB",
      "url": "https://github.com/ethereumjs/ethereumjs-vm/issues/386"
    },
    {
      "type": "ADVISORY",
      "url": "https://github.com/advisories/GHSA-2mw7-wggm-m6w3"
    },
    {
      "type": "PACKAGE",
      "url": "https://github.com/ethereumjs/ethereumjs-vm"
    }
  ],
  "affected": [
    {
      "package": {
        "name": "ethereumjs-vm",
        "ecosystem": "npm",
        "purl": "pkg:npm/ethereumjs-vm"
      },
      "ranges": [
        {
          "type": "SEMVER",
          "events": [
            {
              "introduced": "0"
            },
            {
              "last_affected": "2.4.0"
            }
          ]
        }
      ],
      "database_specific": {
        "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2018/11/GHSA-2mw7-wggm-m6w3/GHSA-2mw7-wggm-m6w3.json"
      }
    }
  ],
  "schema_version": "1.6.0",
  "severity": [
    {
      "type": "CVSS_V3",
      "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
    }
  ]
}